src/daft-plan/src/physical_plan.rs

```diff
@@ -272,27 +363,73 @@ impl PhysicalPlan {
             })
             | Self::HashJoin(HashJoin { left, right, .. })
             | Self::SortMergeJoin(SortMergeJoin { left, right, .. }) => {
-                left.approximate_size_bytes().and_then(|left_size| {
-                    right
-                        .approximate_size_bytes()
-                        .map(|right_size| left_size + right_size)
-                })
+                // assume a Primary-key + Foreign-Key join which would yield the max of the two tables
+                let left_stats = left.approximate_stats();
+                let right_stats = right.approximate_stats();
+
+                ApproxStats {
+                    lower_bound_rows: 0,
+                    upper_bound_rows: left_stats
+                        .upper_bound_rows
+                        .and_then(|l| right_stats.upper_bound_rows.map(|r| l.max(r))),
+                    lower_bound_bytes: 0,
+                    upper_bound_bytes: left_stats
+                        .upper_bound_bytes
+                        .and_then(|l| right_stats.upper_bound_bytes.map(|r| l.max(r))),
+                }
             }
             // TODO(Clark): Approximate post-aggregation sizes via grouping estimates + aggregation type.
-            Self::Aggregate(_) => None,
+            Self::Aggregate(Aggregate { input, groupby, .. }) => {
+                let input_stats = input.approximate_stats();
+                // TODO we should use schema inference here
+                let est_bytes_per_row_lower =
+                    input_stats.lower_bound_bytes / (input_stats.lower_bound_rows.max(1));
+                let est_bytes_per_row_upper = input_stats
+                    .upper_bound_bytes
+                    .and_then(|bytes| input_stats.upper_bound_rows.map(|rows| bytes / rows))
+                    .unwrap_or(100);
```

Comment:
Is this a good default? Might want to leave this as `None` until we can do schema-based estimates.

Reply:
removed unwrap!